python/lsst/meas/modelfit/optimizer/optimizer.cc

```diff
@@ -0,0 +1,207 @@
+// -*- lsst-c++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2013 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "pybind11/pybind11.h"
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+
+#include "lsst/pex/config/pybind11.h"
+
+#include "lsst/meas/modelfit/optimizer.h"
+#include "lsst/meas/modelfit/Model.h"
+#include "lsst/meas/modelfit/Likelihood.h"
+#include "lsst/meas/modelfit/Prior.h"
+#include "lsst/afw/table/Catalog.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace modelfit {
+namespace {
+
+using PyOptimizerObjective = py::class_<OptimizerObjective, std::shared_ptr<OptimizerObjective>>;
+using PyOptimizerControl = py::class_<OptimizerControl, std::shared_ptr<OptimizerControl>>;
+using PyOptimizerHistoryRecorder =
+    py::class_<OptimizerHistoryRecorder, std::shared_ptr<OptimizerHistoryRecorder>>;
+using PyOptimizer = py::class_<Optimizer, std::shared_ptr<Optimizer>>;
+
+PyOptimizerObjective declareOptimizerObjective(py::module & mod) {
+    PyOptimizerObjective cls(mod, "OptimizerObjective");
+    cls.def_readonly("dataSize", &OptimizerObjective::dataSize);
+    cls.def_readonly("parameterSize", &OptimizerObjective::parameterSize);
+    cls.def_static("makeFromLikelihood", &OptimizerObjective::makeFromLikelihood,
+                   "likelihood"_a, "prior"_a=py::none());
```

Comment:
And now that I think about it, we should probably use `nullptr` instead of either of those.

I'll add the comment.